src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import dao.CsvDataReader;
+import dao.CsvDataReaderImpl;
+import dao.CsvDataWriter;
+import dao.CsvDataWriterImpl;
+import db.Storage;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import service.CalculateBalance;
+import service.CalculateBalanceImpl;
+import service.ReportGeneratorImpl;
+
+public class Main {
+    public static void main(String[] args) {
+        String inputFilePath = "src/main/resources/Fruits Data.csv";
+        String outputFilePath = "src/main/resources/Report.csv";
+
+        CsvDataReader csvDataReader = new CsvDataReaderImpl();
+        List<FruitTransaction> fruitTransactions = csvDataReader.readDataFromFile(inputFilePath);
+
+        CalculateBalance calculateBalanceOfFruits = new CalculateBalanceImpl(new Storage());
+        Map<String, Integer> fruitQuantities = calculateBalanceOfFruits
+                .calculateBalance(fruitTransactions);
+
+        ReportGeneratorImpl reportGenerator = new ReportGeneratorImpl();
+        String report = reportGenerator.generateReport(fruitQuantities);
```

Comment:
Sorry, but I cannot get idea why we should use additional Map<String, Integer> fruitQuantities when we have storage.
As I see it seems just to be duplicate.
It's not good to have 2 sources of truth in project.
Please remove fruitQuantities from solution. Handlers will change storage state and then report generator should go to storage and take data from there, not from additional map that duplicates storage

Also it will reduce amount of arguments in constructors and methods

Reply:
Got it. Thanks